src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/AutoplayVuap.java

```diff
@@ -52,27 +37,25 @@
   private final By closeButtonSelector;
   private final By progressBarSelector;
   private final By replayOverlaySelector;
+  private final By videoSelector;
 
   private boolean mobile;
 
-  public AutoplayVuap(WikiaWebDriver driver, String slot, String adIframeId) {
-    this(driver, slot, By.id(adIframeId), false);
-  }
-
-  public AutoplayVuap(WikiaWebDriver driver, String slot, By adIframeSelector, Boolean mobile) {
+  public AutoplayVuap(WikiaWebDriver driver, String slotSelector, By adIframeSelector, Boolean mobile) {
     this.driver = driver;
     this.wait = new Wait(driver);
 
-    this.slot = slot;
-    this.imaBridgeSelector = By.cssSelector("#" + slot + " .video-player iframe[src*='imasdk']");
+    this.slotSelector = slotSelector;
+    this.imaBridgeSelector = By.cssSelector(String.format("#%s %s", slotSelector, IMA_BRIGDE_SELECTOR));
     this.adIframeSelector = adIframeSelector;
     this.mobile = mobile;
 
-    speakerSelector = By.cssSelector(String.format(SPEAKER_SELECTOR_FORMAT, slot));
-    pauseOverlaySelector = By.cssSelector(String.format(PAUSE_BUTTON_SELECTOR_FORMAT, slot));
-    closeButtonSelector = By.cssSelector(String.format(CLOSE_BUTTON_SELECTOR_FORMAT, slot));
-    progressBarSelector = By.cssSelector(String.format(CURRENT_TIME_SELECTOR_FORMAT, slot));
-    replayOverlaySelector = By.cssSelector(String.format(REPLAY_BUTTON_SELECTOR_FORMAT, slot));
+    speakerSelector = By.cssSelector(String.format("%s %s", slotSelector, SPEAKER_SELECTOR));
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "%s %s" 6 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)